api/envoy/api/v2/core/base.proto

```diff
@@ -134,14 +134,14 @@ enum RequestMethod {
 // Header name/value pair.
 message HeaderValue {
   // Header name.
-  string key = 1 [(validate.rules).string.min_bytes = 1];
+  string key = 1 [(validate.rules).string = {min_bytes: 1, max_bytes: 16384}];
 
   // Header value.
   //
   // The same :ref:`format specifier <config_access_log_format>` as used for
   // :ref:`HTTP access logging <config_access_log>` applies here, however
   // unknown header values are replaced with the empty string instead of `-`.
-  string value = 2;
+  string value = 2 [(validate.rules).string.max_bytes = 16384];
```

Comment:
Given that this is a format string, which could pull in the value of other request or response headers, is this adequate?

Reply:
I don't see a way to do arbitrary values.  You can definitely go over the 2^32 limit with the semi-arbitrary metadata, and per htuch's suggestion it'd be good to have run-time validation as well, but I think this has value as a first pass.